### PR TITLE
refactor: drop `TuiDestroyService`

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/addon-mobile/directives/ripple/ripple.providers.ts [hunk 1]
```diff
@@ -69,7 +67,7 @@ export const TUI_RIPPLE_PROVIDERS: Provider[] = [
     },
     {
         provide: TUI_RIPPLE_END,
-        deps: [TUI_TOUCH_SUPPORTED, TUI_IS_IOS, ElementRef, TuiDestroyService],
+        deps: [TUI_TOUCH_SUPPORTED, TUI_IS_IOS, ElementRef],
```
Comment (vladimirpotekhin) [inline projects/addon-mobile/directives/ripple/ripple.providers.ts:70]: `destroy$` still in `useFacrory` args

File: projects/addon-doc/directives/scroll-into-view/scroll-into-view.directive.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export class TuiScrollIntoViewLinkDirective {
                 switchMap(() => this.scroll$),
                 debounceTime(750),
                 filter(shallWe => shallWe && !!tuiGetElementObscures(el)),
-                takeUntil(inject(TuiDestroyService, {self: true})),
+                takeUntilDestroyed(inject(DestroyRef)),
```
Comment (vladimirpotekhin) [inline projects/addon-doc/directives/scroll-into-view/scroll-into-view.directive.ts:22]: here and in many other places `DestroyRef` is unnecessary, just `takeUntilDestroyed()`

Comment (splincode) [reply projects/addon-doc/directives/scroll-into-view/scroll-into-view.directive.ts]: fixed

File: projects/addon-mobile/directives/ripple/ripple.providers.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-mobile/directives/ripple/ripple.providers.ts:71]: fixed

File: projects/addon-mobile/components/sheet/directives/sheet-stop/sheet-stop.directive.ts [hunk 3]
```diff
@@ -31,15 +29,15 @@ export class TuiSheetStopDirective {
                 map(([above, dragged]) => !above && !dragged),
                 filter(Boolean),
                 throttleTime(100),
-                takeUntil(destroy$),
+                takeUntilDestroyed(destroy$),
```
Comment (vladimirpotekhin) [inline projects/addon-mobile/components/sheet/directives/sheet-stop/sheet-stop.directive.ts:32]: `destroy$` is unnecessary

Comment (splincode) [reply projects/addon-mobile/components/sheet/directives/sheet-stop/sheet-stop.directive.ts:32]: We have nested scope

image

File: projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts [hunk 4]
```diff
@@ -139,7 +140,7 @@ export class TuiMobileCalendarComponent implements AfterViewInit {
 
     constructor() {
         inject(TUI_VALUE_STREAM)
-            .pipe(takeUntil(this.destroy$))
+            .pipe(takeUntilDestroyed(this.destroy$))
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts:143]: Could it be just `takeUntilDestroyed()` ?

File: projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts [hunk 5]
```diff
@@ -22,12 +23,12 @@ import {TuiSheetService} from '../../sheet.service';
     templateUrl: './sheets-host.template.html',
     styleUrls: ['./sheets-host.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts:26]: Drop empty line

File: projects/addon-mobile/directives/ripple/ripple.directive.ts [hunk 6]
```diff
@@ -71,7 +71,7 @@ export class TuiRippleDirective {
                         map(() => ripple),
                     );
                 }),
-                takeUntil(this.destroy$),
+                takeUntilDestroyed(this.destroy$),
```
Comment (nsbarsukov) [inline projects/addon-mobile/directives/ripple/ripple.directive.ts:74]: Could it be just `takeUntilDestroyed()` ?

File: projects/addon-mobile/directives/touchable/touchable.directive.ts [hunk 7]
```diff
@@ -49,7 +49,7 @@ export class TuiTouchableDirective {
                         tuiTypedFromEvent(element, 'touchend'),
                     ).pipe(take(1)),
                 ),
-                takeUntil(this.destroy$),
+                takeUntilDestroyed(this.destroy$),
```
Comment (nsbarsukov) [inline projects/addon-mobile/directives/touchable/touchable.directive.ts:52]: just `takeUntilDestroyed()` ?

File: projects/cdk/directives/native-validator/native-validator.directive.ts [hunk 8]
```diff
@@ -1,24 +1,21 @@
...
-    ],
+    providers: [tuiProvide(NG_VALIDATORS, TuiNativeValidatorDirective, true)],
 })
-export class TuiNativeValidatorDirective implements Validator {
+export class TuiNativeValidatorDirective implements Validator, OnDestroy {
     private readonly zone = inject(NgZone);
     private readonly host: HTMLInputElement = inject(ElementRef).nativeElement;
-    private readonly destroy$ = inject(TuiDestroyService, {self: true});
     private control?: AbstractControl;
+    private readonly destroy$ = new Subject<void>();
```
Comment (nsbarsukov) [inline projects/cdk/directives/native-validator/native-validator.directive.ts:18]:
```suggestion
    private readonly destroy$ = new ReplaySubject(1);
```

File: projects/cdk/directives/overscroll/overscroll.directive.ts [hunk 9]
```diff
@@ -25,7 +32,7 @@ export class TuiOverscrollDirective {
             .pipe(
                 filter(() => this.enabled),
                 tuiZonefree(this.zone),
-                takeUntil(this.destroy$),
+                takeUntilDestroyed(this.destroy$),
```
Comment (nsbarsukov) [inline projects/cdk/directives/overscroll/overscroll.directive.ts:35]:
```suggestion
                takeUntilDestroyed(),
```

File: projects/cdk/directives/overscroll/overscroll.directive.ts [hunk 10]
```diff
@@ -69,7 +76,7 @@ export class TuiOverscrollDirective {
                     );
                 }),
                 tuiZonefree(this.zone),
-                takeUntil(this.destroy$),
+                takeUntilDestroyed(this.destroy$),
```
Comment (nsbarsukov) [inline projects/cdk/directives/overscroll/overscroll.directive.ts:79]:
```suggestion
                takeUntilDestroyed(),
```

File: projects/core/components/alert/alert.component.ts [hunk 11]
```diff
@@ -17,7 +23,7 @@ import {TUI_ALERT_POSITION} from './alert.tokens';
     templateUrl: './alert.template.html',
     styleUrls: ['./alert.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/core/components/alert/alert.component.ts:26]: Empty line

File: projects/core/components/expand/expand.component.ts [hunk 12]
```diff
@@ -34,15 +35,15 @@ const LOADER_HEIGHT = 48;
     templateUrl: './expand.template.html',
     styleUrls: ['./expand.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/core/components/expand/expand.component.ts:38]: Drop empty line

File: projects/core/components/hints-host/hints-host.component.ts [hunk 13]
```diff
@@ -18,23 +18,23 @@ import {takeUntil} from 'rxjs';
     // So that we do not force OnPush on custom hints
     // eslint-disable-next-line @angular-eslint/prefer-on-push-component-change-detection
     changeDetection: ChangeDetectionStrategy.Default,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/core/components/hints-host/hints-host.component.ts:21]: Empty line

File: projects/core/components/tooltip/tooltip.component.ts [hunk 14]
```diff
@@ -7,15 +7,15 @@ import {
...
+import {TUI_IS_MOBILE} from '@taiga-ui/cdk';
 import {TuiHintHoverDirective, TuiHintOptionsDirective} from '@taiga-ui/core/directives';
 
 @Component({
     selector: 'tui-tooltip',
     templateUrl: './tooltip.template.html',
     styleUrls: ['./tooltip.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/core/components/tooltip/tooltip.component.ts:18]: empty line

File: projects/experimental/components/tooltip/tooltip.component.ts [hunk 15]
```diff
@@ -23,7 +23,7 @@ import {TUI_TOOLTIP_OPTIONS} from './tooltip.options';
     templateUrl: './tooltip.template.html',
     styleUrls: ['./tooltip.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/experimental/components/tooltip/tooltip.component.ts:26]: Empty line

File: projects/kit/components/checkbox/checkbox.component.ts [hunk 16]
```diff
@@ -25,7 +21,7 @@ import {TUI_CHECKBOX_OPTIONS} from './checkbox.options';
     template: '',
     styleUrls: ['./checkbox.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [TuiDestroyService],
+
```
Comment (nsbarsukov) [inline projects/kit/components/checkbox/checkbox.component.ts:24]: Empty line

File: projects/kit/components/input-number/input-number.component.ts [hunk 17]
```diff
@@ -97,7 +97,7 @@ export class TuiInputNumberComponent
     protected numberFormat = TUI_DEFAULT_NUMBER_FORMAT;
     protected readonly controller = inject(TUI_TEXTFIELD_WATCHED_CONTROLLER);
     protected readonly numberFormat$ = inject(TUI_NUMBER_FORMAT)
-        .pipe(tuiWatch(this.cdr), takeUntil(this.destroy$))
+        .pipe(tuiWatch(this.cdr), takeUntilDestroyed(this.destroy$))
```
Comment (nsbarsukov) [inline projects/kit/components/input-number/input-number.component.ts:100]:
```suggestion
        .pipe(tuiWatch(this.cdr), takeUntilDestroyed())
```

File: projects/kit/components/routable-dialog/routable-dialog.component.ts [hunk 18]
```diff
@@ -32,7 +37,7 @@ export default class TuiRoutableDialogComponent {
                         this.route.snapshot.data['dialogOptions'],
                     ),
                 ),
-                takeUntil(this.destroy$),
+                takeUntilDestroyed(this.destroy$),
```
Comment (nsbarsukov) [inline projects/kit/components/routable-dialog/routable-dialog.component.ts:40]:
```suggestion
                takeUntilDestroyed(),
```

File: projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts (hunk 4 above)
Comment (splincode) [reply projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts]: fixed

Comment (splincode) [reply projects/addon-mobile/components/mobile-calendar/mobile-calendar.component.ts]: fixed

File: projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts (hunk 5 above)
Comment (splincode) [reply projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts]: fixed

File: projects/addon-mobile/directives/ripple/ripple.directive.ts (hunk 6 above)
Comment (splincode) [reply projects/addon-mobile/directives/ripple/ripple.directive.ts]: fixed

File: projects/cdk/directives/native-validator/native-validator.directive.ts (hunk 8 above)
Comment (splincode) [reply projects/cdk/directives/native-validator/native-validator.directive.ts]: fixed

File: projects/cdk/directives/overscroll/overscroll.directive.ts (hunk 9 above)
Comment (splincode) [reply projects/cdk/directives/overscroll/overscroll.directive.ts]: fixed

File: projects/cdk/directives/overscroll/overscroll.directive.ts (hunk 10 above)
Comment (splincode) [reply projects/cdk/directives/overscroll/overscroll.directive.ts]: fixed

File: projects/core/components/alert/alert.component.ts (hunk 11 above)
Comment (splincode) [reply projects/core/components/alert/alert.component.ts]: fixed

File: projects/kit/components/routable-dialog/routable-dialog.component.ts (hunk 18 above)
Comment (splincode) [reply projects/kit/components/routable-dialog/routable-dialog.component.ts]: fixed

File: projects/kit/components/input-number/input-number.component.ts (hunk 17 above)
Comment (splincode) [reply projects/kit/components/input-number/input-number.component.ts]: fixed

File: projects/kit/components/checkbox/checkbox.component.ts (hunk 16 above)
Comment (splincode) [reply projects/kit/components/checkbox/checkbox.component.ts]: fixed

File: projects/experimental/components/tooltip/tooltip.component.ts (hunk 15 above)
Comment (splincode) [reply projects/experimental/components/tooltip/tooltip.component.ts]: fixed

File: projects/core/components/expand/expand.component.ts (hunk 12 above)
Comment (splincode) [reply projects/core/components/expand/expand.component.ts]: fixed

File: projects/core/components/hints-host/hints-host.component.ts (hunk 13 above)
Comment (splincode) [reply projects/core/components/hints-host/hints-host.component.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**